### DISCUSSION TITLE
FSI at high Reynolds number: Vector or scalar field velocity variable?

### DISCUSSION THREAD

**Discussion by Eilloo** [idaholab/moose#23567] [Q&A General] [answered]
Hi all,

I am working towards running FSI simulations at high Reynolds numbers:

I have had some success using the fsi module at low Reynolds numbers, where the velocity is a scalar field variable (ie: one variable per component), and also running CFD only simulations in FE at high Reynolds numbers. In the latter case, SUPG stabilisation is required, and the velocity is a vector variable.

To combine the two approaches, I believe I need to either:
1. Allow the interface kernels to accept a vector velocity variable, without changing the structural variables to be vectors too. This idea is described near the end of [this](https://github.com/idaholab/moose/discussions/23484) thread.
2. Add a kernel to implement SUPG stabilisation, which accepts the velocity as three separate components. and applies a stabilisation term to each appropriately.

I have done some digging, and anticipate the following obstacles with each approach:

In the first case, the main problem is that the class template for the interface kernel expects only one variable type. 
The more I look into the code, the more I think it is not straightforward to add a second type, and change everything this feeds as well. Indeed, I'm encountering situations where the type of one object would be dependent on the type of another, which so far is troublesome in C++.
The bottom line here is the implications of messing with the class template seem complex and relatively far-reaching in terms of other changes I would need to make.

In the second case, the current implementation of SUPG stabilisation in the INS module is tied to material classes, which expect a vector velocity variable. My initial thinking was to extract all of the equations which go into calculating this term and collect them in a new, ordinary kernel which does the calculations according to a scalar field velocity. This way I would avoid redefining large chunks of these built-in material classes.
However, I noticed that the current SUPG kernel is not an ordinary kernel, but an 'ADStabilizedKernel'. I've not found much documentation for this, but in the header file there is a comment noting: 'Called before forming the residual for an element'. 
I would like to know whether stabilisation terms such as this can be included is ordinary kernels with a residual contribution as proposed above, or whether some special treatment is required?

Phew! Apologies for the information dump - I wanted to be as clear as possible about the current state of affairs. Presently, I am thinking that option 2 is likely easier to achieve.

In addition to the specific question about stabilisation kernels, I am hoping to get some thoughts thoughts from anyone who's more involved/familiar with these modules. 
I would welcome any insight regarding which is more feasible to achieve, or even suggestions of some altogether simpler solution.

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

Sorry the delay.
Scalar field velocity variables will be easier to deal with. There are simply many more objects you can use with scalar fields over vector fields.

Guillaume

**Comment by Eilloo** [comment]
No problem, thanks for the advice!

**Comment by lindsayad** [comment]
In https://github.com/idaholab/moose/pull/24204 I am introducing an interface kernel that interfaces the component treatment of displacements in the tensor mechanics module with the vector velocity treatment of INSAD
